### ISSUE TITLE
Minor improvement of subscriber registration

### ISSUE THREAD

**Issue by jaclas** [spinettaro/delphi-event-bus#52]
Please add the parameter _ARaiseExcIfEmpty : Boolean_ to the _RegisterSubscriber_() method to pass it on to the _FindSubscriberMethods_() method. 
At the moment, in the _RegisterSubscriber_() method, when calling the _FindSubscriberMethods_() method, this parameter is hardcoded to True, it should not be so, see:

```
procedure TEventBus.RegisterSubscriber<T>(ASubscriber: TObject); <== please add ARaiseExcIfEmpty param
var
  LSubscriberClass: TClass;
  LSubscriberMethods: TArray<TSubscriberMethod>;
  LSubscriberMethod: TSubscriberMethod;
begin
  FMultiReadExclWriteSync.BeginWrite;

  try
    LSubscriberClass := ASubscriber.ClassType;
    LSubscriberMethods := TSubscribersFinder.FindSubscriberMethods<T>(LSubscriberClass, True); <==== HARDCODED
    for LSubscriberMethod in LSubscriberMethods do Subscribe<T>(ASubscriber, LSubscriberMethod);
  finally
    FMultiReadExclWriteSync.EndWrite;
  end;
end;
```

I have a problem with this because I have a base class that supports caching and I want to call _RegisterSubscriberForEvents_() in this base class (not to do this in many inherited classes). But because not every descendant cache class uses events, I get an **EObjectHasNoSubscriberMethods** exception in runtime.

And of course a change is necessary in these two methods

```
procedure TEventBus.RegisterSubscriberForChannels(ASubscriber: TObject); <== ARaiseExcIfEmpty param
begin
  RegisterSubscriber<ChannelAttribute>(ASubscriber);
end;

procedure TEventBus.RegisterSubscriberForEvents(ASubscriber: TObject); <== ARaiseExcIfEmpty param
begin
  RegisterSubscriber<SubscribeAttribute>(ASubscriber);
end;

```

**Comment by wxinix** [comment]
I created a pull-request to address your issue:

Added two new methods:
- IEvenBus.SilentRegisterSubscriberForChannels(ASubscriber: TObject);
- IEvenBus.SilentRegisterSubscriberForEvents(ASubscriber: TObject);

I **added** new methods instead of **overloading** the existing ones, for two reasons:
1. Compatible with existing code
2. Make the intention clear, that is, silently register the subscriber without throwing exceptions when the subscriber object does not have subscriber methods defined.

This should fit your needs.  You might want to check out my fork:  https://github.com/wxinix/delphi-event-bus
